fastval.go

```diff
@@ -381,6 +381,10 @@ func (val FastVal) compareTime(other FastVal) int {
 	thisTime := val.AsTime()
 	otherTime := other.AsTime()
 
+	if thisTime == nil || otherTime == nil {
+		return -1
+	}
```

Comment:
not sure about this. not symmetrical. maybe more consistent if we consider nil time to be the smaller one?

Reply:
That's a valid point... Although in some cases where comparing 2 invalid DATE() fastvals for equality will return "true", but that's probably just due to equality of nil values.
With the parser returning error when DATE expression is invalid should cut down on that situation.

Fixed